code/modules/admin/server_verbs.dm

```diff
@@ -17,6 +17,87 @@
+		return
+
+	if (alert("Are you sure you want to shutdown the server? Only somebody with remote access to the server can turn it back on.", "Shutdown Server?", "Cancel", "Shutdown Server", "Cancel.") != "Shutdown Server")
+		return
+
+	if (!SSticker)
+		if (alert("The game ticker does not exist, normal checks will be bypassed.", "Continue Shutting Down Server?", "Cancel", "Continue Shutting Down Server", "Cancel.") != "Continue Shutting Down Server")
+			return
+	else
+		var/required_state_message = "The server must be in either pre-game or post-game and the start/end must be delayed to shutdown the server."
+		if (SSticker.current_state != GAME_STATE_PREGAME && SSticker.current_state != GAME_STATE_FINISHED)
+			to_chat(usr, "<span class='danger'>[required_state_message] The round is not in the lobby or endgame state.</span>")
+			return
+		if ((SSticker.current_state == GAME_STATE_PREGAME && going) || (SSticker.current_state == GAME_STATE_FINISHED && !SSticker.delay_end))
+			to_chat(usr, "<span class='danger'>[required_state_message] The round start/end is not delayed.</span>")
+			return
+
+	to_chat(usr, "<span class='danger'>Alert: Delayed confirmation required. You will be asked to confirm again in 30 seconds.</span>")
+	message_admins("[ADMIN_TPMONTY(usr)] Is considering shutting down the server. Admins with +server may abort this by pressing the shutdown server button again.")
+	shuttingdown = usr.ckey
+
+	sleep(30 SECONDS)
+
+	if (!shuttingdown || shuttingdown != usr.ckey)
+		return
+
+	if (!usr?.client)
+		message_admins("[ADMIN_TPMONTY(usr)] left the server before they could finish confirming they wanted to shutdown the server.")
+		shuttingdown = null
+		return
+
+	if (alert("ARE YOU SURE YOU WANT TO SHUTDOWN THE SERVER? ONLY SOMEBODY WITH REMOTE ACCESS TO THE SERVER CAN TURN IT BACK ON.", "Shutdown Server?", "Cancel", "Yes! Shutdown The Server!", "Cancel.") != "Yes! Shutdown The Server!")
+		message_admins("[ADMIN_TPMONTY(usr)] decided against shutting down the server.")
+		shuttingdown = null
+		return
+
+	to_chat(world, "<span class='danger'>Server shutting down in 30 seconds!</span> <span class='notice'>Initiated by: [usr.key]</span>")
+	message_admins("[ADMIN_TPMONTY(usr)] Is shutting down the server. Admins with +server may abort this by pressing the shutdown server button again within 30 seconds.")
+
+	sleep(31 SECONDS) //to give the admins that final second to hit the confirm button on the cancel prompt.
+
+	if (!shuttingdown)
+		to_chat(world, "<span class='notice'>Server shutdown was aborted</span>")
+		return
+
+	if (shuttingdown != usr.ckey) //somebody cancelled but then somebody started again.
+		return
+
+	to_chat(world, "<span class='danger'>Server shutting down.</span> <span class='notice'>Initiated by: [shuttingdown]</span>")
+
+	if (GLOB.tgs)
+		var/datum/tgs_api/TA = GLOB.tgs
+		TA.EndProcess()
```

Comment:
@Cyberboss this restarts the server, it doesn't shut it down like the name implies.

What should I put here if I want to shut the server down?